src/utils/token_to_fiat_oracle.ts

```diff
@@ -0,0 +1,137 @@
+import type { FiatOracle } from './fiat_oracle';
+import type { CoinGeckoPriceRequestParams, CoinGeckoPriceResponseData } from './coingecko_types';
+import { TokenFiatPair, TokenFiatRate } from './token_fiat_price';
+import { FiatID, fiatIdToThirdParty, TokenID, tokenIdToThirdParty } from './fiat_oracle_types';
+
+const pollingIntervalMilliseconds = 1000 * 60 * 15; // 15 min
+
+function cachePairFilterFactory(pair: TokenFiatPair) {
+	return (rate: TokenFiatRate) => rate.fiat === pair.fiat && rate.token === pair.token;
+}
+
+/*
+ * The CoinGecko wrapper does not support browser.
+ * Here it is the reported feature request for supporting
+ * the browser platform by just removing node dependencies.
+ *
+ * https://github.com/miscavage/CoinGecko-API/issues/19
+ *
+ * When the report is taken and fixed, please:
+ * 1. Import the CoinGecko wrapper.
+ * 2. Define a CoinGecko client as private readonly at the
+ *   constructor arguments. So we can test it.
+ * 3. On each call to the API, replace the fetch for the actual
+ *   API call.
+ *
+ * Examples
+ *
+ * 1. import CoinGecko from 'coingecko-api'
+ * 2. constructor(private readonly client = new CoinGecko()) {}
+ * 3. const response = await this.client.ping() // for the ping() function
+ *    const response = await this.client.simple.price(params) // for getting the price of a currency
+ */
+
+export class TokenToFiatOracle implements FiatOracle {
+	private cachedPrices: TokenFiatRate[] = [];
+	private cacheTimestamp = 0;
+	private syncPromise?: Promise<void>;
+
+	/**
+	 * @param {FiatID[]} fiats The currency IDs to quote tokens in
+	 * @param {TokenID[]} tokens The token ID of the requested coin
+	 * @param {number} cacheLifespan Milliseconds that has to pass until the cache is invalid
+	 * @param {@link fetch} _fetch The standard fetch method. Defined here for testing propuses
+	 * @return {TokenToFiatOracle} A class representing the oracle for the selected currencies
+	 */
+	constructor(
+		private readonly fiats: FiatID[] = ['USD'],
+		private readonly tokens: TokenID[] = ['ARWEAVE'],
+		private readonly cacheLifespan = pollingIntervalMilliseconds,
+		private readonly _fetch = fetch
+	) {}
+
+	private get currentlyFetchingPrice(): boolean {
+		return !!this.syncPromise;
+	}
+
+	private get tokensCSV(): string {
+		return this.tokens.map(tokenIdToThirdParty).join(',');
+	}
+
+	private get fiatCSV(): string {
+		return this.fiats.map(fiatIdToThirdParty).join(',');
+	}
+
+	private get shouldRefreshCacheData(): boolean {
+		const currentTimestamp = Date.now();
+		const deltaTimestamp = currentTimestamp - this.cacheTimestamp;
```

Comment:
`this.cacheTimestamp` doesn't seem to be set anywhere other than the default 0. This is resulting in always refreshing the cache, irregardless of how fresh or stale it is.

Reply:
The issue here was the wrong naming. It's now `this.cacheLifespan`, lifespan is constant :)